client/component/src/DomainConfig.tsx

```diff
@@ -191,12 +209,26 @@ function DomainConfig(props: DomainConfigProps) {
             dataDomain={dataDomain}
             value={sliderDomain}
             scaleType={scaleType ?? ScaleType.Linear}
-            onChangeMin={(val) => onCustomDomainChange([val, customDomain[1]])}
-            onChangeMax={(val) => onCustomDomainChange([customDomain[0], val])}
+            onChangeMin={(val) =>
+              onCustomDomainChange([val, memoizedCustomDomain[1]])
+            }
+            onChangeMax={(val) =>
+              onCustomDomainChange([memoizedCustomDomain[0], val])
+            }
             {...histogram}
           />
         )}
       </DomainTools>
+      <label>
+        <input
+          type="checkbox"
+          checked={lockDomain}
+          onChange={() => {
+            setLockDomain(!lockDomain);
```

Comment:
This is wrong - I think it should be `setLockDomain((l) => !l)`

Reply:
Changed setter